app/scripts/editor/token_highlighter.js

```diff
@@ -0,0 +1,108 @@
+    var updateTimer = null;
+    editor.on('change', function(){
+        clearTimeout(updateTimer);
+        if(!editor.TokenHighlighter.highlightingTokens){
+            updateTimer = setTimeout(editor.TokenHighlighter.highlightTokens, 500);
+        }
+    });
+    editor.commands.addCommand({
+        name: 'manPageShortcut',
+        bindKey: {win: 'Ctrl-I', mac: 'Command-M'},
+        exec: function(editor){
+            var pos = editor.getCursorPosition();
+            var token = editor.session.getTokenAt(pos.row, pos.column);
+            editor.TokenHighlighter.eventHandler(token);
+        }
+    });
+    editor.on('click', this.onClick);
+}
+
+(function () {
+    var self = this;
+    self.markers = [];
+    self.tokenStore = {};
+
+    this.highlightTokens = function () {
+        this.highlightingTokens = true;
+
+        var session = this.editor.session;
+        var docLength = session.getLength();
+        var manPageTokens = this.manPageTokens;
+        var row = 1;
+        var tokenStart = 0;
+
+        var searchAndHighlight = function(token){
+            if(token.type === 'identifier'){
+                manPageTokens.get(token.value, function(results){
+                    var i = 0;
+                    if(results.length && results[i].name === token.value){
+                        var result = results[i];
+                        while(result && result.section !== 2 && result.section !== 3){
+                            result = results[++i];
+                        }
+                        if(result){
+                            self.tokenStore[token.value] = result;
+                            var range = new Range(row, tokenStart, row, tokenStart+token.value.length);
+                            self.markers.push(session.addMarker(range, 'token-highlight', 'text'));
+                        }
+                    }
+                });
+            }
+            tokenStart += token.value.length;
+        };
+        self.markers.forEach(function(marker){
+            session.removeMarker(marker);
+        });
+        self.markers = [];
+
+        for (row = 1; row <= docLength; row++) {
+            var tokens = session.getTokens(row);
+            tokenStart = 0;
+            tokens.forEach(searchAndHighlight);
+        }
+        this.highlightingTokens = false;
+    };
+
+    this.onClick = function (e) {
+        var editor = e.editor;
+        var r = editor.renderer;
+        var canvasPos = r.rect || (r.rect = r.scroller.getBoundingClientRect());
+        var offset = (e.clientX + r.scrollLeft - canvasPos.left - r.$padding) / r.characterWidth;
+        var row = Math.floor((e.clientY + r.scrollTop - canvasPos.top) / r.lineHeight);
+        var column = Math.round(offset);
+        var screenPos = {row: row, column: column, side: offset - column > 0 ? 1 : -1};
+        var token = editor.session.getTokenAt(screenPos.row, screenPos.column);
+        if(token && token.type === 'identifier') {
```

Comment:
Yes, they will tokenized although it will be a syntax error in itself if a C variable clashes with a man page token.